src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -80,23 +88,59 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
                 displayMetadata.DataTypeName = DataType.Html.ToString();
             }
 
+
+            var localizedType = context.Key.ContainerType ?? context.Key.ModelType;
```

Comment:
`localizedType` isn't meaningful. See `modelType` for the same expression elsewhere though `containerType` might also work.